File: livingrimoire start here/LivinGrimoire java/src/Skills/special_skills/DiEngager.java
package Skills.special_skills;

import Auxiliary_Modules.DrawRndDigits;
import Auxiliary_Modules.LGFIFO;
import Auxiliary_Modules.TimeUtils;
import Auxiliary_Modules.TrgMinute;
import LivinGrimoire.Skill;

public class DiEngager extends Skill {
    // makes the AI burp n  times per hour at random times
    private int burpsPerHour = 2;
    private TrgMinute trgMinute = new TrgMinute(0);
    private String skillToEngage = "unknown";
    private DrawRndDigits draw = new DrawRndDigits();
    private LGFIFO<Integer> burpMinutes = new LGFIFO<>();
    public DiEngager(int burpsPerHour, String skillToEngage) {
        super();
        if((burpsPerHour >0)&& (burpsPerHour <60)){
            this.burpsPerHour = burpsPerHour;
        }
        for (int i = 1; i < 60; i++) {
            draw.addElement(i);
        }
        for (int i = 0; i < burpsPerHour; i++) {
            burpMinutes.add(draw.draw());
        }
        this.skillToEngage = skillToEngage;
    }

    public void setSkillToEngage(String skillToEngage) {
        this.skillToEngage = skillToEngage;
    }

    @Override
    public void input(String ear, String skin, String eye) {
        // night? do not burp
        if (TimeUtils.partOfDay().equals("night")){
            return;
        }
        // reset burps
        if (trgMinute.trigger()){
            burpMinutes.clear();
            draw.reset();
            for (int i = 0; i < burpsPerHour; i++) {
                burpMinutes.add(draw.draw());
            }
            return;
        }
        // burp
        int nowMinutes = TimeUtils.getMinutesAsInt();
        if(burpMinutes.contains(nowMinutes)){
            burpMinutes.removeItem(nowMinutes);
            this.kokoro.toHeart.put(skillToEngage, "engage");
            return;
        }
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/skills/special/DiEngager.kt
package skills.special

import auxiliary_modules.*
import livinGrimoire.DiSkillV2

class DiEngager(burpsPerHour: Int, skillToEngage: String) : DiSkillV2() {
    // makes the AI burp n  times per hour at random times
    private var burpsPerHour = 2
    private val trgMinute: TrgMinute = TrgMinute(0)
    private var skillToEngage = "unknown"
    private val draw: DrawRndDigits = DrawRndDigits()
    private val burpMinutes: LGFIFO<Int> = LGFIFO()
    private val pl: TimeUtils = TimeUtils()

    init {
        if (burpsPerHour in 1..59) {
            this.burpsPerHour = burpsPerHour
        }
        for (i in 1..59) {
            draw.addElement(i)
        }
        for (i in 0 until burpsPerHour) {
            burpMinutes.add(draw.draw())
        }
        this.skillToEngage = skillToEngage
    }

    fun setSkillToEngage(skillToEngage: String) {
        this.skillToEngage = skillToEngage
    }

    override fun input(ear: String, skin: String, eye: String) {
        // night? do not burp
        if (pl.partOfDay().equals("night")) {
            return
        }
        // reset burps
        if (trgMinute.trigger()) {
            burpMinutes.clear()
            draw.reset()
            for (i in 0 until burpsPerHour) {
                burpMinutes.add(draw.draw())
            }
            return
        }
        // burp
        val nowMinutes: Int = pl.minutesAsInt
        if (burpMinutes.contains(nowMinutes)) {
            burpMinutes.removeItem(nowMinutes)
            this.kokoro.toHeart.put(skillToEngage, "engage")
        }
    }
}